cogs/Announce.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python
+
+"""
+Koala Bot Announce feature
+Created by: Bill Cao
+"""
+
+# Built-in/Generic Imports
+import asyncio
+
+# Libs
+import math
+from typing import Optional, Tuple
+
+import discord
+from discord.ext import commands
+from utils.KoalaUtils import extract_id
+from utils import KoalaColours
+import time
+
+# constants
+ANNOUNCE_SEPARATION_DAYS = 30
+SECONDS_IN_A_DAY = 86400
+TIMEOUT_TIME = 60.0
+MAX_MESSAGE_LENGTH = 2000
+
+import KoalaBot
+
+
+def announce_is_enabled(ctx):
+    """
+    A command used to check if the guild has enabled announce
+    e.g. @commands.check(announce_is_enabled)
+
+    :param ctx: The context of the message
+    :return: True if enabled or test, False otherwise
+    """
+    try:
+        result = KoalaBot.check_guild_has_ext(ctx, "Announce")
+    except PermissionError:
+        result = False
+
+    return result or (str(ctx.guild) == KoalaBot.TEST_USER and KoalaBot.is_dpytest)
+
+
+async def wait_for_message(bot: discord.Client, ctx: commands.Context, timeout: float = TIMEOUT_TIME) -> Tuple[
+    Optional[discord.Message], Optional[discord.TextChannel]]:
+    """
+        Wraps bot.wait_for with message event, checking that message author is the original context author. Has default
+        timeout of 60 seconds.
+        :param bot: Koala Bot client
+        :param ctx: Context of the original command
+        :param timeout: Time to wait before raising TimeoutError
+        :return: If a message (msg) was received, returns a tuple (msg, None). Else returns (None, ctx.channel)
+        """
+    try:
+        msg = await bot.wait_for('message', timeout=timeout, check=lambda message: message.author == ctx.author)
+    except (Exception, TypeError):
+        return None, ctx.channel
+    if not msg:
+        return msg, ctx.channel
+    return msg, None
+
+
+class Announce(commands.Cog):
+    """
+        A discord.py cog to allow announcements to certain roles.
```

Comment:
```suggestion
        Send DM announcements to specified people and roles.
```
This will show up in `k!help`, so worth making it slightly more user friendly (i know Twitch Alert is bad for this, I'll fix it later)